spec/design/skel/Structs_B.thy

```diff
@@ -0,0 +1,24 @@
+(*
+ * Copyright 2024, Proofcraft Pty Ltd
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ *)
+
+(* Generic data types shared by spec and abstract. *)
+
+chapter "Common, Generic Data Types"
+
+theory Structs_B
+imports Arch_Structs_B
+begin
+
+arch_requalify_types (H)
+  arch_tcb_flag
+
+arch_requalify_consts (H)
+  archTcbFlagToWord
+
+#INCLUDE_SETTINGS keep_constructor = tcb_flag
```

Comment:
Do you happen to know why we tell it to keep constructors for `tcb_flag` when we're importing `TcbFlag`? Does a `tcb_flag` actually exist anywhere? It doesn't seem the most easily graspable interface, I'm glad you managed to figure it out.

Reply:
Another case of the translator being magic; `TcbFlag` is quietly translated to `tcb_flag`.